programs/mpl-core/src/plugins/verified_creators.rs

```diff
@@ -0,0 +1,205 @@
+use std::collections::{BTreeMap, HashSet};
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use solana_program::{program_error::ProgramError, pubkey::Pubkey};
+
+use crate::{error::MplCoreError, plugins::PluginType, state::Authority};
+
+use super::{Plugin, PluginValidation, PluginValidationContext, ValidationResult};
+
+/// The creator on an asset and whether or not they are verified.
+#[derive(Clone, BorshSerialize, BorshDeserialize, Debug, PartialEq, Eq, Hash)]
+pub struct VerifiedCreatorsSignature {
+    address: Pubkey,
+    verified: bool,
+}
+
+/// Structure for storing verified creators, often used in conjunction with the Royalties plugin
+#[derive(Clone, BorshSerialize, BorshDeserialize, Debug, Eq, PartialEq)]
+pub struct VerifiedCreators {
+    /// A list of signatures
+    signatures: Vec<VerifiedCreatorsSignature>,
+}
+
+struct SignatureChangeIndices {
+    /// Indices of added signatures on new_verified_creators
+    added: Vec<u8>,
+    /// Indices of changed signatures on new_verified_creators
+    changed: Vec<u8>,
+    /// Indices of removed signatures on verified_creators
+    removed: Vec<u8>,
+}
+
+fn calculate_signature_changes(
+    new_verified_creators: &VerifiedCreators,
+    verified_creators: Option<&VerifiedCreators>,
+) -> Result<SignatureChangeIndices, ProgramError> {
+    let mut existing_map: BTreeMap<Pubkey, &VerifiedCreatorsSignature> = BTreeMap::new();
+    if let Some(verified_creators) = verified_creators {
+        for sig in verified_creators.signatures.iter() {
+            existing_map.insert(sig.address, sig);
+        }
+    }
+
+    let new_signatures: HashSet<_> = new_verified_creators
+        .signatures
+        .iter()
+        .map(|sig| sig.address)
+        .collect();
+
+    if new_verified_creators.signatures.len() != new_signatures.len() {
+        // Ensure there are no duplicate signatures
+        solana_program::msg!("Verified creators: Rejected");
+        return Err(MplCoreError::InvalidPluginSetting.into());
+    }
+
+    let mut result = SignatureChangeIndices {
+        added: Vec::new(),
```

Comment:
Is it with using with_capacity() here? Or is there nothing known at that point.

Reply:
nothing is known however we can probably assume that the vec will always be size 0 or 1 given that there can only ever be one authority.